### ISSUE TITLE
test_ctm_energy_explicit_gradient_matches_fd fails only on macOS Full tests CI

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#369]
## Symptom

\`tests/test_ctm_energy_explicit.py::test_ctm_energy_explicit_gradient_matches_fd\` fails on the **macOS 3.12** \`Full tests\` job and passes on **Linux 3.11/3.12**.

The test asserts:

\`\`\`python
assert max_rel_err < 0.05, (
    f\"Gradient relative error {max_rel_err:.4e} > 0.05 ({checked} elements checked)\"
)
\`\`\`

with \`eps=1e-5\`, comparing \`jax.grad\` of \`ctm_energy_explicit\` vs central-difference FD on 20 elements of a random D=2 d=2 site tensor at chi=4.

## Diagnostic context

- Marked \`@pytest.mark.slow\`; runs in the post-merge \`Full tests\` matrix only — required CI checks (which run \`-m core\`) are unaffected.
- The implicit-AD analog \`test_ctm_energy_implicit.py::test_ctm_energy_implicit_gradient_matches_fd\` **passes on the same macOS job**. The drift is specific to the explicit-AD chain.
- Test uses \`projector_method=\"eigh\"\` (line 51 of \`test_ctm_energy_explicit.py\`).
- The macOS Full tests job often hits \`exit 137\` (OOM) before pytest writes the failure summary section, so the actual \`max_rel_err\` value at failure isn't in CI logs. Reproduce locally on Apple Silicon to capture it.

## Likely root cause

Apple's Accelerate framework (used by JAX on macOS) has known precision differences from OpenBLAS in the LAPACK \`syevd\` / eigh path used by the explicit-AD projectors. The \`eigh\` backward divides by eigenvalue gaps and is sensitive to projector-direction precision in a way that the implicit-AD GMRES path is not.

## Test source (file:line for repro)

- \`tests/test_ctm_energy_explicit.py:27\` — \`test_ctm_energy_explicit_gradient_matches_fd\`
- \`tests/test_ctm_energy_explicit.py:51\` — explicit \`projector_method=\"eigh\"\`
- \`tests/test_ctm_energy_explicit.py:90\` — eps=1e-5, n_check=20
- \`tests/test_ctm_energy_explicit.py:108\` — \`max_rel_err < 0.05\` assertion

## Repro on Apple Silicon

\`\`\`bash
JAX_PLATFORMS=cpu uv run pytest tests/test_ctm_energy_explicit.py::test_ctm_energy_explicit_gradient_matches_fd -v
\`\`\`

Capture the actual \`max_rel_err\` value to tell whether the drift is marginal (< 0.10) or significant (> 0.20) — that decides between defensive tolerance bump and a real numerical investigation.

## Triage options once reproduced

1. **\`max_rel_err < 0.10\` on macOS:** bump tolerance to 0.10 with a platform-aware \`pytest.mark.skipif\` or per-platform constant. Most pragmatic.
2. **\`max_rel_err > 0.20\` on macOS:** investigate the eigh backward — likely needs a multiplet-aware projector or tighter eigenvector regularization for Accelerate's eigh output.
3. **AD ≠ FD by orders of magnitude:** drop \`eps\` to 1e-4 (FD truncation may be amplifying); if still bad, the explicit-AD path is genuinely producing a different gradient on macOS and warrants the deeper investigation in (2).

## Background: why now

Surfaced as the last remaining failure in the \`Full tests\` matrix after the #354 cleanup waves (#355–#368). Not a regression from any recent PR — the test has been failing on macOS at least since the 2026-04-30 03:49 main run, but was masked by other macOS failures (\`test_python_loop_matches_existing_ctm\` deleted in #367; the \`test_optimize_lbfgs_runs\` OOM addressed by #364). With those gone, this is the only macOS-only test failure left.
